src/webgpu/api/validation/createBindGroup.spec.ts

```diff
@@ -743,3 +744,52 @@ g.test('storage_texture,mip_level_count')
       });
     }, mipLevelCount !== 1);
   });
+
+g.test('buffer,usage')
+  .desc(
+    `
+    Test that the buffer usage contains 'uniform' if the BindGroup entry defines
+    buffer and it's type is 'uniform'.
+  `
+  )
+  .params(u =>
+    u //
+      // If usage0 and usage1 are the same, the usage being test is a single usage. Otherwise, it's
+      // a combined usage.
+      .combine('usage0', kBufferUsages)
+      .combine('usage1', kBufferUsages)
+      .unless(
+        ({ usage0, usage1 }) =>
+          ((usage0 | usage1) & (GPUConst.BufferUsage.MAP_READ | GPUConst.BufferUsage.MAP_WRITE)) !==
+          0
+      )
+  )
+  .fn(async t => {
+    const { usage0, usage1 } = t.params;
+
+    const usage = usage0 | usage1;
+
+    const bindGroupLayout = t.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.COMPUTE,
+          buffer: { type: 'uniform' },
+        },
+      ],
+    });
+
+    const buffer = t.device.createBuffer({
+      size: 1024,
```

Comment:
nit: I'd create a smaller buffer to reduce memory pressure slightly since the size doesn't matter.
```suggestion
      size: 4,
```

Reply:
Fixed.